dotcom-rendering/src/web/layouts/DecideLayout.tsx

```diff
@@ -33,25 +34,43 @@ export const DecideLayout = ({ CAPI, NAV }: Props): JSX.Element => {
 	// TODO we can probably better express this as data
 	switch (display) {
 		case Display.Immersive: {
-			if (design === Design.Interactive) {
-				return (
-					<InteractiveImmersiveLayout
-						CAPI={CAPI}
-						NAV={NAV}
-						format={format}
-						palette={palette}
-					/>
-				);
+			switch (design) {
+				case Design.Interactive: {
+					// TBD: Remove when we migrate legacy content
+					switch (CAPI.isLegacyInteractive) {
+						case true: {
+							return (
+								<FullPageInteractiveLayout
+									CAPI={CAPI}
+									NAV={NAV}
+									format={format}
+									palette={palette}
+								/>
+							);
+						}
+						default: {
```

Comment:
Minor, but might be clearer to match on `false` here. Or maybe even do an if/else as the nested switches can be a bit harder to read sometimes.